tool/grind/generate_deprecations.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:dart_style/dart_style.dart';
+import 'package:yaml/yaml.dart';
+
+void updateDeprecationFile(File yamlFile) {
```

Comment:
It would be more consistent with other libraries to define the task here directly.